src/account/address.cpp

```diff
@@ -1,36 +1,56 @@
 #include "account/address.h"
-#include <sodium.h>
 #include "bits.h"
 #include "bech32/bech32.h"
+#include "params.h"
+
+#include <sodium.h>
+#include <stdexcept>
 
 Address::Address(bytes const &publicKey) :
         m_pk(publicKey),
         m_bech32Address()
-{}
+{
+    if (publicKey.size() != crypto_sign_PUBLICKEYBYTES)
+        throw std::length_error(ERROR_MSG_KEY_BYTES_SIZE);
+}
 
 Address::Address(std::string const &bech32Address) :
         m_pk(),
         m_bech32Address(bech32Address)
-{}
+{
+    if (util::bech32::decode(bech32Address).first != hrp)
+        throw std::invalid_argument(ERROR_MSG_BECH32);
```

Comment:
👍 for defining constants for error messages. Perhaps move them to a file called `errors.h`, as well (instead of keeping them in `params.h`)?

Reply:
Thank you ! :smile: 
That's a good point, I'll move them in a separate file.